include/mbedtls/ssl_ciphersuites.h

```diff
@@ -371,10 +371,18 @@ typedef enum {
 #endif
 
 /* TLS 1.2 or 1.3 key exchanges using ECDH(E) */
+#if defined(MBEDTLS_KEY_EXCHANGE_SOME_ECDH_OR_ECDHE_1_2_ENABLED) || \
+    (defined(MBEDTLS_SSL_TLS1_3_KEY_EXCHANGE_MODE_EPHEMERAL_ENABLED) || \
+    defined(MBEDTLS_SSL_TLS1_3_KEY_EXCHANGE_MODE_PSK_EPHEMERAL_ENABLED)) && \
+    !defined(PSA_WANT_ALG_FFDH)
+#define MBEDTLS_KEY_EXCHANGE_SOME_ECDH_OR_ECDHE_ANY_ENABLED
+#endif
+
+/* TLS 1.2 or 1.3 DH key exchanges: ECDH(E) or FFDH */
 #if defined(MBEDTLS_KEY_EXCHANGE_SOME_ECDH_OR_ECDHE_1_2_ENABLED) || \
     defined(MBEDTLS_SSL_TLS1_3_KEY_EXCHANGE_MODE_EPHEMERAL_ENABLED) || \
     defined(MBEDTLS_SSL_TLS1_3_KEY_EXCHANGE_MODE_PSK_EPHEMERAL_ENABLED)
-#define MBEDTLS_KEY_EXCHANGE_SOME_ECDH_OR_ECDHE_ANY_ENABLED
+#define MBEDTLS_KEY_EXCHANGE_SOME_DH_ANY_ENABLED
```

Comment:
What exactly are we trying to capture here? According to the comment it's ECDH(E) of FFDH but:
- if that's really the case we'd need to include the DHE-RSA 1.2 key exchange
- but I'm not sure what would be the purpose.

Perhaps we're trying to capture "a key exchange that may use the `xxdh` fields in the handshake structure (for 1.2, when `USE_PSA` is enabled)? I can see how that could be useful. But then I suggest naming it `XXDH_ANY` as a reference to those fields, and a longer comment. (Note: they're only named `xxdh` in #7858 which is not merged yet.)

(Minor, unrelated: can simplify the condition by using `MBEDTLS_SSL_TLS1_3_KEY_EXCHANGE_MODE_SOME_EPHEMERAL_ENABLED`.)